bin/pasta_show_maintainers.py

```diff
@@ -86,72 +76,68 @@ def get_small(title, counts):
-        results = list()
-        if args.filter != None and args.query == 'entries':
-            results.append([title, 'Status', 'LoC in list', 'Total LoC', 'Byte count',  'In list(%)', 'Other directories'])
-            for item in counts_filt:
-                if any(all_maintainers.subsystems[item].status):
-                    results.append([item, all_maintainers.subsystems[item].status[0].value,counts_filt[item][0], counts[item][0], counts_filt[item][1], 
-                    round((counts_filt[item][0]/counts[item][0])*100, 2), ', '.join(dirs[item].difference(dirs_filt[item]))]) 
-                else:
-                    results.append([item, '', counts_filt[item][0], counts[item][0], counts_filt[item][1], round((counts_filt[item][0]/counts[item][0])*100, 2)])
-        elif args.filter != None and args.query == 'maintainers':
-            results.append([title, 'LoC in list', 'Total LoC', 'Byte count',  'In list(%)', 'Entries'])
-            for item in counts_filt:
-                results.append([item, counts_filt[item][0], counts[item][0], counts_filt[item][1], round((counts_filt[item][0]/counts[item][0])*100, 2), ', '.join(maintainers_subs[item]) ])
-        elif args.query == 'entries':
-            results.append([title, 'Status', 'LoC', 'Byte count'])
-            for item in counts:
-                if any(all_maintainers.subsystems[item].status):
-                    results.append([item, all_maintainers.subsystems[item].status[0].value, counts[item][0], counts[item][1]]) 
-                else:
-                    results.append([item, '', counts[item][0], counts[item][1]])
-        else:
-            results.append([title, 'LoC', 'Byte count', 'Entries'])
-            for item in counts:
-                results.append([item, counts[item][0], counts[item][1], ', '.join(maintainers_subs[item])])
+        percentage = lambda item : round((counts_filt[item][0]/counts[item][0])*100, 2)
+        status = lambda item : all_maintainers.subsystems[item].status[0].value if any(all_maintainers.subsystems[item].status) else ''
         
+        cases = dict()
+        cases['entries', True] =[lambda item : [item, counts_filt[item][0], counts[item][0], counts_filt[item][1], percentage(item), status(item), ', '.join(dirs[item].difference(dirs_filt[item]))], [args.query.capitalize(), 'LoC in list', 'Total LoC', 'Byte count',  'In list(%)','Status', 'Other directories']]
+        cases['entries', False] =[lambda item : [item, counts[item][0], counts[item][1], status(item)], [args.query.capitalize(), 'LoC', 'Byte count','Status']]
+        cases['maintainers', True] =[lambda item : [item, counts_filt[item][0], counts[item][0], counts_filt[item][1], percentage(item), ', '.join(maintainers_subs[item])], [args.query.capitalize(), 'LoC in list', 'Total LoC', 'Byte count',  'In list(%)', 'Entries']]
+        cases['maintainers', False] =[lambda item : [item, counts[item][0], counts[item][1]], [args.query.capitalize(), 'LoC', 'Byte count']]
+
```

Comment:
Now, look what are duplicate lambda functions and what is the configurable part here? If you split it accordingly, you got a much nicer solution.

Reply:
I actually meant a dictionary for 'field_name' => function, so 'LoC' => lambda item: counts[item][0]... etc.

Reply:
>     * I did not understand "Other directories"
> 
> 
> How about:
> 
>     * RELEVANT: list of the most general file patterns such that all files with that pattern are included in the relevant list and are in the entry.
> 
>     * IRRELEVANT: list of the most general file pattern such that all files under that directory are excluded from the relevant list, but are in the entry.
> 
> 
> Then, I think that should cover all cases...

Hello, 
Other directories are showing the directories that are excluded in the filter list for each of the entries. Would it be meaningful to display some aggregation of these directories in this case?